src/needless_features.rs

```diff
@@ -0,0 +1,49 @@
+//! Checks for usage of nightly features that have simple stable equivalents
+//!
+//! This lint is **warn** by default
+
+use rustc::lint::*;
+use rustc_front::hir::*;
+
+use utils::{span_lint};
+
+declare_lint! {
+    pub AS_SLICE,
+    Warn,
+    "as_slice is not stable and can be replaced by & v[..]\
+see https://github.com/rust-lang/rust/issues/27729"
+}
+
+declare_lint! {
+    pub AS_MUT_SLICE,
+    Warn,
+    "as_mut_slice is not stable and can be replaced by &mut v[..]\
+see https://github.com/rust-lang/rust/issues/27729"
+}
+
+
+#[derive(Copy,Clone)]
+pub struct NeedlessFeaturesPass;
+
+impl LintPass for NeedlessFeaturesPass {
+    fn get_lints(&self) -> LintArray {
+        lint_array!(AS_SLICE,AS_MUT_SLICE)
+    }
+}
+
+impl LateLintPass for NeedlessFeaturesPass {
+    fn check_expr(&mut self, cx: &LateContext, expr: &Expr) {
+        if let ExprMethodCall(ref name, _, _) = expr.node {
+            if name.node.as_str() == "as_slice" {
```

Comment:
We should then match against all of those :)

You can also use [`get_attrs`](http://manishearth.github.io/rust-internals-docs/rustc/middle/ty/struct.ctxt.html#method.get_attrs) (after getting the defid via  `cx.tcx.def_map`) and check if the unstable attribute is there.